### PR TITLE
test: use shell-test (avoid system shell)

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: test/functional/helpers.lua [hunk 1]
```diff
@@ -302,6 +302,10 @@ end
 local function nvim_feed(input)
   while #input > 0 do
     local written = module.request('nvim_input', input)
+    if written == nil then
+      module.assert_alive()
```
Comment (blueyed) [inline test/functional/helpers.lua]: `module.assert_alive` does not call `assert`, but returns something only.

**Comment by blueyed** [comment]
AppVeyor failure (but on MINGW):
```
[  FAILED  ] test/functional\eval\environ_spec.lua @ 14: environment variables exists() handles empty env variable
test/functional\eval\environ_spec.lua:16: Expected objects to be the same.
Passed in:
(number) 1
Expected:
(number) 0
```
Likely needs to be rebased (anyway)?

File: test/functional/helpers.lua (hunk 1 above)
Comment (justinmk) [reply test/functional/helpers.lua]: `eq()` does an assert. But `return` can be removed.

**Comment by justinmk** [comment]
the new `wildmode_spec` fails sometimes on quickbuild, and also on my local machine. Will investigate.

**Comment by blueyed** [comment]
This one?  (osx, Travis, https://github.com/neovim/neovim/pull/10814)
```
[  ERROR   ] test/functional/ui/wildmode_spec.lua @ 83: 'wildmenu' is preserved during :terminal activity
test/functional/ui/screen.lua:580: Row 1 did not match.
Expected:
  |*                         |
  |                         |
  |                         |
  |define  jump  list  >    |
  |:sign define^             |
Actual:
  |*foo                      |
  |foo                      |
  |foo                      |
  |define  jump  list  >    |
  |:sign define^             |
To print the expect() call that would assert the current screen state, use
screen:snapshot_util(). In case of non-deterministic failures, use
screen:redraw_debug() to show all intermediate screen states.  
stack traceback:
	test/functional/ui/screen.lua:580: in function '_wait'
	test/functional/ui/screen.lua:368: in function 'expect'
	test/functional/ui/wildmode_spec.lua:21: in function 'expect_stay_unchanged'
	test/functional/ui/wildmode_spec.lua:96: in function <test/functional/ui/wildmode_spec.lua:83>
```

**Comment by blueyed** [comment]
@justinmk rebased.

**Comment by justinmk** [comment]
failed on master:

```
[  ERROR   ] test/functional\ui\wildmode_spec.lua @ 84: 'wildmenu' is preserved during :terminal activity
test\functional\ui\screen.lua:587: Row 1 did not match.
Expected:
  |*:sign                    |
  |*define    place          |
  |*jump      undefine       |
  |*list      unplace        |
  |*:sign ^                   |
Actual:
  |*0: !terminal             |
  |*                         |
  |*^                         |
  |*~                        |
  |*                         |
```

**Pull request merged**